### PR TITLE
[fix] torch.multinomial : fix for 0 size dim

### LINKED ISSUE #43768: Multinomial distribution of empty tensor crashes when sampling with replacement on CUDA
## 🐛 Bug

Multinomial distribution of empty tensor weights crashes when sampling with replacement on CUDA. However, on CPU it works as expected. For the context, I use a mask to filter unnecessary rows and sometimes it covers the whole tensor thus I have to deal with empty tensors.

## To Reproduce

Reproduce the behavior (downscaled snippet):

```
device = "cuda" if torch.cuda.is_available() else "cpu"

x = torch.randn(0, 128)
# Works as expected as there is nothing 0x64
torch.multinomial(x, 64, replacement = True)
# Crashes
torch.multinomial(x.to(device), 64, replacement = True)
```

## Environment

Tested on PyTorch both 1.6.0 (Google Colab / CUDA 10.1) and 1.4.0 (Local Machine / CUDA 10.1 & GTX 1080 TI).

## Stack

`THCudaCheck FAIL file=/pytorch/aten/src/THC/generic/THCTensorScatterGather.cu line=95 error=9 : invalid configuration argument
`

cc @ezyang @gchanan @zou3519 @ngimel

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes #43768

TO-DO:
* [x] Add test

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -17797,8 +17797,8 @@ def test(probs, replacement):
             test(z, True)
 
     def test_multinomial_empty(self, device):
-        probs = torch.ones(0, 3)
-        num_samples = 1
+        probs = torch.ones(0, 128, device=device)
+        num_samples = 64
```
Comment (kshitij12345) [inline test/test_torch.py:17801]: Curiously, test passed with
```
probs = torch.ones(0, 3, device=device)
num_samples = 1
```

Comment (DomainFlag) [reply test/test_torch.py:17801]: Is it smth with replacement, as it fails only when num_samples > 1?

Comment (kshitij12345) [reply test/test_torch.py:17801]: Possible. I haven't actually stepped through the exact kernel code where it is failing.

**Comment by kshitij12345** [comment]
@ngimel Please review :)

**Comment by ngimel** [comment]
Thanks for the fix. I think that fixing root cause would be better, otherwise it is probably triggered in other situations.

File: test/test_torch.py (hunk 1 above)
Comment (ngimel) [reply test/test_torch.py:17801]: Test passes with replacement=False, so you have to separately test different replacement modes.

**Comment by ngimel** [comment]
It looks like the error is in  multinomial_kernel_impl, so the fix is ok. Can you also please add missing 
```
AT_CUDA_CHECK(cudaGetLastError());
```
at line 396 of MultinomialKernel.cu?
And test `replacement=True` and `replacement=False`.

File: test/test_torch.py (hunk 1 above)
Comment (kshitij12345) [reply test/test_torch.py:17801]:
```python
>>> torch.multinomial(x, 1, True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: CUDA error: invalid configuration argument
```

The above error comes from 
https://github.com/pytorch/pytorch/blob/58148c85f4c7db372602ed9332fa6ea0ade3e703/aten/src/ATen/native/cuda/MultinomialKernel.cu#L320-L342

For `num_samples > 1`,

```python
>>> torch.multinomial(x, 2, True)
Floating point exception (core dumped)
```

https://github.com/pytorch/pytorch/blob/58148c85f4c7db372602ed9332fa6ea0ade3e703/aten/src/ATen/native/cuda/MultinomialKernel.cu#L364-L383

Here `grid_y` is `0` and thus in the snippet below, we get floating point exception due to Divide By Zero.
https://github.com/pytorch/pytorch/blob/58148c85f4c7db372602ed9332fa6ea0ade3e703/aten/src/ATen/native/cuda/MultinomialKernel.cu#L381

The fix takes care of both these cases.

**Comment by kshitij12345** [comment]
Gentle Ping :)

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**